numpy/lib/function_base.py

```diff
@@ -4154,32 +4155,87 @@ def _median(a, axis=None, out=None, overwrite_input=False):
         return mean(part[indexer], axis=axis, out=out)
 
 
+def _valid_quantile(q):
+    # avoid expensive reductions, relevant for arrays with < O(1000) elements
+    if q.ndim == 1 and q.size < 10:
+        for i in range(q.size):
+            if q[i] < 0.0 or q[i] > 1.0:
+                return False
+    else:
+        # faster than any()
+        if np.count_nonzero(q < 0.0) or np.count_nonzero(q > 1.0):
+            return False
+    return True
+
+
 def percentile(a, q, axis=None, out=None,
                overwrite_input=False, interpolation='linear', keepdims=False):
     """
     Compute the qth percentile of the data along the specified axis.
 
-    Returns the qth percentile(s) of the array elements.
+    The ``q``\ th percentile is synonymous with the `quantile` at ``q/100``.
+
+    Parameters
+    ----------
+    q : real number in range of [0,100] (or sequence of real numbers)
+        Percentile to compute, which must be between 0 and 100 inclusive.
+        This is divided by 100 and passed on to `quantile`.
+
+    All the other parameters have the same meaning as they do in `quantile`.
+
+    See also
+    --------
+    quantile
+    mean
+    median : equivalent to ``percentile(..., 50)``
+    nanpercentile
+
+    Examples
+    --------
+    >>> a = np.array([[10, 7, 4], [3, 2, 1]])
+    >>> a
+    array([[10,  7,  4],
+           [ 3,  2,  1]])
+    >>> np.percentile(a, 50)
+    3.5
+    >>> np.quantile(a, 0.5)
+    3.5
+
+
+    """
+    q = np.true_divide(q, 100)  # handles the asarray for us too
+    if not _valid_quantile(q):
+        raise ValueError("Percentiles must be in the range [0, 100]")
+    return quantile(a, q, axis, out, overwrite_input, interpolation, keepdims)
+
+
+def quantile(a, q, axis=None, out=None,
+             overwrite_input=False, interpolation='linear', keepdims=False):
+    """
+    Compute the `q`th quantile of the data along the specified axis.
+
+    ..versionadded:: 1.14
 
     Parameters
     ----------
     a : array_like
         Input array or object that can be converted to an array.
-    q : float in range of [0,100] (or sequence of floats)
-        Percentile to compute, which must be between 0 and 100 inclusive.
+    q : float in range of [0,1] (or sequence of floats)
+        Quantile to compute, which must be between 0.0 and 1.0 inclusive.
     axis : {int, sequence of int, None}, optional
-        Axis or axes along which the percentiles are computed. The
-        default is to compute the percentile(s) along a flattened
-        version of the array. A sequence of axes is supported since
-        version 1.9.0.
+        Axis or axes along which the quantiles are computed. The
+        default is to compute the quantile(s) along a flattened
+        version of the array.
+        .. versionchanged:: 1.9.0
+           A sequence of axes is supported
```

Comment:
I think we should just make this "A tuple of axes is supported" so that it resembles ufuncs. Sure, a sequence will work, but in general we would prefer everyone use tuples.

Reply:
Good catch